src/Nethermind/Nethermind.Consensus/Producers/TxPoolTxSource.cs

```diff
@@ -57,8 +58,8 @@ public class TxPoolTxSource : ITxSource
         public IEnumerable<Transaction> GetTransactions(BlockHeader parent, long gasLimit)
         {
             long blockNumber = parent.Number + 1;
-            IReleaseSpec releaseSpec = _specProvider.GetSpec(blockNumber);
-            UInt256 baseFee = BaseFeeCalculator.Calculate(parent, releaseSpec);
+            IReleaseSpec specFor1559 = _specProvider.GetSpecFor1559(blockNumber);
+            UInt256 baseFee = BaseFeeCalculator.Calculate(parent, specFor1559);
```

Comment:
Shouldn't we use full spec here?

Reply:
Will address this in the next PR with slotTime in the config.